assignments/this.js

```diff
@@ -1,26 +1,57 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. Implicit binding the this keyword is referring to whatever is to the left of the dot. 
+* 2. Explicit binding uses call, apply and bind to say what a function is going to be. Call makes you indicate each argument individually, apply allows you to use an array and bind gives you a new array you can call later.
+* 3. New binding when used, a new object is created inside of the function. 
+* 4. Window binding is the default use of the this keyword. If nothing is specified, window binding seaches the entire environment. This keyword is used outside of a function.
 *
 * write out a code example of each explanation above
 */
 
-// Principle 1
+// Principle 1 - Window (Global)
 
-// code example for Window Binding
+function(window) {
+	return this;
+}
 
-// Principle 2
+// Principle 2 - Implicit
 
-// code example for Implicit Binding
+const me = {
+	name: 'Erin',
+	sayName: function(){
+		console.log(this.name);
+	}
+};
+me.SayName();
 
-// Principle 3
 
-// code example for New Binding
+// Principle 3 - New
 
-// Principle 4
+function Pet(obj) {
+	this.petName = 'Shadow';
+	this.species = 'cat';
+	this.age = 14;
+	this.whatAmI = function(iAm){
+		return `${this.petName} is a ${species}`
+	}
+}
 
-// code example for Explicit Binding
\ No newline at end of file
+let Toby = new Pet({
+	petName: 'Toby';
+	species: 'dog';
+	age: 17;
+})
```

Comment:
How would you output the whatAmI function after you use the `new` keyword?

Reply:
During my peer review, I realized I never called it. I am going back to fix this now.